src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -71,8 +71,14 @@ public async Task<ExecutionResult> ExecuteAsync(ExecutionOptions options)
                 {
                     using (metrics.Subject("schema", "Initializing schema"))
```

Comment:
Move this using under lock too.

Reply:
That would be inappropriate.  If there is lock contention, it would be because the schema is being initialized.  Without the metric, the time spent waiting for the lock would not be timed or reported in the metrics.  And if there is no lock contention then it does not matter.

Reply:
OK. One can look at it from this side too.

Reply:
True.  But it would seem that you'd want to know why your query froze for a period of time.